jbpm-workitems/jbpm-workitems-email/src/main/java/org/jbpm/process/workitem/email/EmailNotificationPublisher.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.process.workitem.email;
+
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+import org.jbpm.runtime.manager.impl.identity.UserDataServiceProvider;
+import org.kie.api.task.model.Group;
+import org.kie.api.task.model.OrganizationalEntity;
+import org.kie.api.task.model.User;
+import org.kie.internal.task.api.UserInfo;
+import org.kie.internal.utils.NotificationPublisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class EmailNotificationPublisher implements NotificationPublisher {
+
+    private static final Logger logger = LoggerFactory.getLogger(EmailNotificationPublisher.class);
+    private boolean active = Boolean.parseBoolean(System.getProperty("org.jbpm.email.publisher.enabled", "true"));
+    
+    private UserInfo userInfo;
+    private Connection connection;
+    
+    private TemplateManager templateManager = TemplateManager.get();
+    
+    public EmailNotificationPublisher() {
+        this.userInfo = UserDataServiceProvider.getUserInfo();
+        Properties conf = new Properties();
+        try {
+            conf.load(this.getClass().getResourceAsStream("/email.properties")); 
```

Comment:
Having the properties file on classpath, does it mean that for e.g. KIE server I have to edit the WAR file?

Reply:
no, in actual server you should use mail session taken from JNDI, property files is more for embedded use cases where mail sessions are not in JNDI

Reply:
I see, so in that case the second constructor will be used to pass there a connection.